### ISSUE TITLE
Vale gets confused with block and line comments

### ISSUE THREAD

**Issue by edcsnt** [jdkato/txtlint#1020]
### Check for existing issues

- [x] Completed

### Environment

* Fedora Workstation 42
* <https://copr.fedorainfracloud.org/coprs/mczernek/vale/>
* `3.12.0`

### Describe the bug / provide steps to reproduce it

```shell
$ more .vale.ini
IgnoredScopes = text.comment.line

[*]
BasedOnStyles = Vale
$ more a.py
#
# a a

def b():
    """c
    d
    """
$ vale a.py

 a.py
 9:5  error  'a' is repeated!  Vale.Repetition 

✖ 1 error, 0 warnings and 0 suggestions in 1 file.

```

Take note of the reported line of the error: 9, even though the files has only 7 lines. It’s as if Vale’s wrapping around the EOF.

Interestingly, the bug is not triggered in these situations:

* If, in the docstring, the `d` line is removed.
* If, in the header comment, the first empty comment line is deleted.
* If the docstring is removed altogether.
* If the docstring is at the module level, i.e., not inside a function.
